.changeset/violet-dryers-cross.md

```diff
@@ -0,0 +1,14 @@
+---
+"@t3-oss/env-nextjs": minor
+"@t3-oss/env-core": minor
+"@t3-oss/env-nuxt": minor
+---
+
+feat!: added ability to customise schema combination
+
+Combination of schemas can now be customised using the `createFinalSchema` option. This allows further refinement or transformation of the environment variables.
+
+This comes with a type-only breaking change:
+
+- `CreateEnv` now has the signature `CreateEnv<TFinalSchema, TExtends>`, instead of the previous `CreateEnv<TServer, TClient, TShared, TExtends>`.
+  - Previous behaviour can be achieved by using `DefaultCombinedSchema<TServer, TClient, TShared>` as the type for `TFinalSchema`.
```

Comment:
is this a user-breaking change?

Reply:
CreateEnv is exported, so technically yes - in practicality I doubt anyone was relying on it

Reply:
It's exported cause the other packages needs it. I should add some @internal jsdoc annotations to things